fl4health/losses/deep_mmd_loss.py

```diff
@@ -0,0 +1,224 @@
+from typing import Optional, Tuple
+
+import numpy as np
+import torch
+
+
+class ModelLatentF(torch.nn.Module):
+    """Latent space for both domains."""
+
+    def __init__(self, x_in: int, H: int, x_out: int):
+        """Init latent features."""
+        super(ModelLatentF, self).__init__()
+        self.restored = False
+        self.latent = torch.nn.Sequential(
+            torch.nn.Linear(x_in, H, bias=True),
+            torch.nn.Softplus(),
+            torch.nn.Linear(H, H, bias=True),
+            torch.nn.Softplus(),
+            torch.nn.Linear(H, H, bias=True),
+            torch.nn.Softplus(),
+            torch.nn.Linear(H, x_out, bias=True),
+        )
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        """Forward the LeNet."""
+        fealant = self.latent(input)
+        return fealant
+
+
+class DeepMmdLoss(torch.nn.Module):
```

Comment:
I think it's worth recording the paper that this is being taken from somewhere here 🙂